client-gen/src/snapshots/sails_clientgen__generator__tests__basic_works.snap

```diff
@@ -1,36 +1,43 @@
 ---
 source: client-gen/src/generator.rs
-expression: generate(program).unwrap()
+expression: "generate(program, \"Service\").unwrap()"
 ---
 // Code generated by sails-client-gen. DO NOT EDIT.
 use parity_scale_codec::{Decode, Encode};
 #[allow(unused_imports)]
 use sails_rtl::collections::BTreeMap;
-use sails_rtl::{String, *};
 #[allow(unused_imports)]
-use sails_sender::{Call, CreateProgramCall, GStdSender};
-pub trait Service {
-    fn do_this(&mut self, p1: u32, p2: MyParam) -> Call<u16>;
-    fn do_that(&mut self, p1: (u8, u32)) -> Call<u8>;
+use sails_rtl::{
+    calls::RemotingAction, gstd::calls::Args, gstd::calls::Remoting, prelude::*, String,
+};
+pub trait Service<TCallArgs = ()> {
+    #[allow(clippy::type_complexity)]
+    fn do_this(&mut self, p1: u32, p2: MyParam) -> RemotingAction<Remoting, TCallArgs, u16>;
+    #[allow(clippy::type_complexity)]
+    fn do_that(&mut self, p1: (u8, u32)) -> RemotingAction<Remoting, TCallArgs, u8>;
 }
 
 #[derive(Clone)]
-pub struct Client {
-    sender: GStdSender,
+pub struct ServiceClient {
```

Comment:
Let's probably keep the same name as for the trait but move all this code into some `impl` sub-module. Another option is adding some `Impl` suffix like ServiceImpl. Don't know which way is more idiomatic for rustecians.